miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,702 @@
+        return self.set_property("forbid_mode", value_str)
+
+    @command(
+        click.argument("start_hr", type=int),
+        click.argument("start_min", type=int),
+        click.argument("end_hr", type=int),
+        click.argument("end_min", type=int),
+    )
+    def set_dnd(self, start_hr: int, start_min: int, end_hr: int, end_min: int):
+        """Set do-not-disturb.
+
+        :param int start_hr: Start hour
+        :param int start_min: Start minute
+        :param int end_hr: End hour
+        :param int end_min: End minute
+        """
+        start_int = int(timedelta(hours=start_hr, minutes=start_min).total_seconds())
+        end_int = int(timedelta(hours=end_hr, minutes=end_min).total_seconds())
+        return self._set_dnd(start_int, end_int, active=True)
+
+    @command()
+    def disable_dnd(self):
+        """Disable do-not-disturb."""
+        # The current do not disturb is read back for a better user expierence,
+        # as start/end time must be set together with enabled=False
+        try:
+            current_dnd_str = self.get_property_by(**_MAPPING["forbid_mode"])[0][
+                "value"
+            ]
+            current_dnd_dict = json.loads(current_dnd_str)
+        except Exception:
+            # In case reading current DND back fails, DND is disabled anyway
+            return self._set_dnd(0, 0, active=False)
+        return self._set_dnd(
+            current_dnd_dict["time"][0], current_dnd_dict["time"][1], active=False
+        )
+
+    @command(click.argument("water_level", type=EnumType(WaterLevel)))
+    def set_water_level(self, water_level: WaterLevel):
+        """Set water_level."""
+        return self.set_property("water_level", water_level.value)
+
+    @command(click.argument("double_clean", type=bool))
+    def set_double_clean(self, double_clean: bool):
+        """Set double clean (True/False)."""
+        return self.set_property("double_clean", double_clean)
+
+    @command(click.argument("edge_sweep", type=bool))
+    def set_edge_sweep(self, edge_sweep: bool):
+        """Set edge_sweep (True/False)."""
+        return self.set_property("edge_sweep", edge_sweep)
+
+    @command(click.argument("lidar_collision", type=bool))
+    def set_lidar_collision(self, lidar_collision: bool):
+        """Set lidar collision (True/False).."""
+        return self.set_property("lidar_collision", lidar_collision)
+
+    @command()
+    def start_dust(self) -> None:
+        """Start base dust collection."""
+        return self.call_action_by(8, 6)
+
+    @command(click.argument("voice", type=str))
+    def set_voice_unknown(self, voice: str) -> None:
+        """Set voice.
+
+        FIXME: the syntax of voice is unknown
+        """
+        return self.call_action_by(8, 12, voice)
+
+    @command()
+    def reset_filter_life(self) -> None:
+        """Reset filter life."""
+        return self.call_action_by(10, 1)
```

Comment:
Please convert the actions to use `call_action(str)` and declare the actions in the mapping like: https://github.com/rytilahti/python-miio/pull/867/files#diff-2f2ea69d0a6bf409d623cea17e4744ce7b60739d4c29b79166a8c1979660bfb3R36-R42

The `call_action_by` is rather targeted for `miiocli` usage where there is no known symbol for it available.